content/blog/kamal-2-multi-server-deployment-complete-guide/index.md

```
@@ -0,0 +1,340 @@
+categories: ["Development", "Rails", "DevOps"]
+canonical_url: "https://jetthoughts.com/blog/kamal-2-multi-server-deployment-complete-guide/"
+slug: "kamal-2-multi-server-deployment-complete-guide"
+cover_image: "cover.png"
+metatags:
+  image: cover.png
+cover_image_alt: "Dark technical cover for Kamal 2 Multi-Server Deployment. JetThoughts ENGINEERING brand mark, Ruby on Rails 2026 pill, low-poly ruby gem, stat chips for hosts, roles, and rolling boot, MULTI-SERVER status indicator."
+---
+
+Add a second host under `servers:` and run `kamal deploy`. If your proxy config has `ssl: true`, the deploy dies before Kamal opens a single SSH connection:
+
+```
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)